src/components/strategies/edit/EditStrategyBudgetContent.tsx

```diff
@@ -1,277 +0,0 @@
-          buyLowWithdrawalBudgetUsd: strategyEventData.buyBudgetUsd,
-          sellHighWithdrawalBudget: strategyEventData.sellBudget,
-          sellHighWithdrawalBudgetUsd: strategyEventData.sellBudgetUsd,
-        })
-      : carbonEvents.strategyEdit.strategyDeposit({
-          ...strategyEventData,
-          strategyId: strategy.id,
-          buyBudget: buyBalance,
-          buyBudgetUsd,
-          sellBudget: buyBalance,
-          sellBudgetUsd,
-          buyLowDepositBudget: strategyEventData.buyBudget,
-          buyLowDepositBudgetUsd: strategyEventData.buyBudgetUsd,
-          sellHighDepositBudget: strategyEventData.sellBudget,
-          sellHighDepositBudgetUsd: strategyEventData.sellBudgetUsd,
-        });
-  };
-
-  const hasChanged = strategyHasChanged(strategy, order0, order1);
-
-  const isDisabled = (form: HTMLFormElement) => {
-    if (approval.isError) return true;
-    if (!hasChanged) return true;
-    if (!form.checkValidity()) return true;
-    if (form.querySelector('.error-message')) return true;
-    if (!form.querySelector('.warning-message')) return false;
-    const checkbox = form.querySelector<HTMLInputElement>('.approve-warnings');
-    return !!checkbox && !checkbox.checked;
-  };
-
-  const submit = (e: FormEvent<HTMLFormElement>) => {
-    e.preventDefault();
-    if (isDisabled(e.currentTarget)) return;
-    if (type === 'withdraw') {
-      const withdrawAll = !+order0.budget && !+order1.budget;
-      if (withdrawAll) {
-        openWithdrawModal();
-      } else {
-        depositOrWithdrawFunds();
-      }
-    } else {
-      if (approval.approvalRequired) {
-        openModal('txConfirm', {
-          approvalTokens: approval.tokens,
-          onConfirm: depositOrWithdrawFunds,
-          buttonLabel: `Confirm Deposit`,
-          eventData: {
-            ...strategyEventData,
-            productType: 'strategy',
-            approvalTokens: approval.tokens,
-            buyToken: strategy.base,
-            sellToken: strategy.quote,
-            blockchainNetwork: provider?.network?.name || '',
-          },
-          context: 'depositStrategyFunds',
-        });
-      } else {
-        depositOrWithdrawFunds();
-      }
-    }
-  };
-
-  const getMarginalOption = (order: OrderCreate, initialBudget: string) => {
-    if (isOverlapping) return order.marginalPrice;
-    if (order.budget === initialBudget) return undefined;
-    if (order.marginalPriceOption) return order.marginalPriceOption;
-    return MarginalPriceOptions.reset;
-  };
-
-  const { deleteStrategy } = useDeleteStrategy();
-
-  const openWithdrawModal = () => {
-    openModal('withdrawOrDelete', {
-      onWithdraw: depositOrWithdrawFunds,
-      onDelete: () =>
-        deleteStrategy(strategy, setIsProcessing, () =>
-          carbonEvents.strategyEdit.strategyDelete({
-            strategyId: strategy.id,
-            ...strategyEventData,
-          })
-        ),
-    });
-  };
```

Comment:
This flow is missing from strategy budget edit. If withdrawing everything -> prompt user to instead delete the strategy